src/sinks/gcp/chronicle_unstructured.rs

```diff
@@ -0,0 +1,521 @@
+    gcp::{GcpAuthConfig, GcpAuthenticator},
+    http::{HttpClient, HttpError},
+    sinks::{
+        gcs_common::{
+            config::{healthcheck_response, GcsRetryLogic},
+            service::GcsResponse,
+            sink::GcsSink,
+        },
+        util::{
+            encoding::{
+                Encoder, EncodingConfig, EncodingConfigWithFramingAdapter, StandardEncodings,
+                StandardEncodingsWithFramingMigrator,
+            },
+            metadata::{RequestMetadata, RequestMetadataBuilder},
+            partitioner::KeyPartitioner,
+            request_builder::EncodeResult,
+            BatchConfig,
+            BulkSizeBasedDefaultBatchSettings,
+            Compression,
+            //Compressor,
+            RequestBuilder,
+            TowerRequestConfig,
+        },
+        Healthcheck,
+    },
+    template::{Template, TemplateParseError},
+    tls::{TlsConfig, TlsSettings},
+};
+
+const NAME: &str = "gcp_chronicle_unstructured";
+
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub))]
+pub enum GcsHealthcheckError {
+    #[snafu(display("log_type template parse error: {}", source))]
+    LogTypeTemplate { source: TemplateParseError },
+
+    #[snafu(display("Endpoint not found"))]
+    NotFound,
+}
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Region {
+    Eu,
+    Us,
+    Asia,
+}
+
+impl Region {
+    /// Each region has a it's own endpoint.
+    fn endpoint(self) -> &'static str {
+        match self {
+            Region::Eu => "https://europe-malachiteingestion-pa.googleapis.com",
+            Region::Us => "https://malachiteingestion-pa.googleapis.com",
+            Region::Asia => "https://asia-southeast1-malachiteingestion-pa.googleapis.com",
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct ChronicleUnstructuredConfig {
+    pub endpoint: Option<String>,
+    pub region: Option<Region>,
+    pub customer_id: String,
+    #[serde(flatten)]
+    pub auth: GcpAuthConfig,
+    #[serde(default)]
+    pub batch: BatchConfig<BulkSizeBasedDefaultBatchSettings>,
+    #[serde(flatten)]
+    pub encoding: EncodingConfigWithFramingAdapter<
+        EncodingConfig<StandardEncodings>,
+        StandardEncodingsWithFramingMigrator,
+    >,
+    #[serde(default)]
+    pub request: TowerRequestConfig,
+    #[serde(default)]
+    compression: Compression,
+    pub tls: Option<TlsConfig>,
+    pub log_type: Template,
+    #[serde(
+        default,
+        deserialize_with = "crate::serde::bool_or_struct",
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    acknowledgements: AcknowledgementsConfig,
+}
+
+inventory::submit! {
+    SinkDescription::new::<ChronicleUnstructuredConfig>(NAME)
+}
+
+impl GenerateConfig for ChronicleUnstructuredConfig {
+    fn generate_config() -> toml::Value {
+        toml::from_str(indoc! {r#"
```

Comment:
We should get a consensus as a team and commit to doing it one way or another.

Reply:
I'd say we defer that to another issue since that would affect more sinks. Personally I don't have any strong opinions about either style. I can see @StephenWakely's argument, it has helped me a couple of times to catch deserialization issues as well

Reply:
Yep - 100% not needing to be handled here!